packages/babel-core/test/parse.js

```diff
@@ -17,19 +17,40 @@ function fixture(...args) {
 
 describe("parse", function () {
   it("should parse using configuration from .babelrc when a filename is provided", function () {
-    const input = fs.readFileSync(fixture("input.js"), "utf8");
-    const output = require(fixture("output"));
+    const input = fs.readFileSync(
+      fixture("filename-provided", "input.js"),
+      "utf8",
+    );
+    const output = require(fixture("filename-provided", "output.json"));
 
     const result = parse(input, {
-      filename: fixture("input.js"),
+      filename: fixture("filename-provided", "input.js"),
+      cwd: fixture(),
+    });
+    expect(JSON.parse(JSON.stringify(result))).toEqual(output);
+  });
+
+  it("should parse using the specified sourceFileName", function () {
+    const input = fs.readFileSync(
+      fixture("sourceFileName-provided", "input.js"),
+      "utf8",
+    );
+    const output = require(fixture("sourceFileName-provided", "output.json"));
+
+    const result = parse(input, {
+      filename: fixture("sourceFileName-provided", "input.js"),
+      sourceFileName: "input.js",
       cwd: fixture(),
     });
     expect(JSON.parse(JSON.stringify(result))).toEqual(output);
   });
 
   it("should parse using passed in configuration", function () {
-    const input = fs.readFileSync(fixture("input.js"), "utf8");
-    const output = require(fixture("output.json"));
+    const input = fs.readFileSync(
```

Comment:
@nicolo-ribaudo this tests the no-"unknown" handling because no `filename`, `filenameRelative` or `sourceFileName` is passed.